CHANGELOG.md

```diff
@@ -184,6 +184,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Add support for query profiler with concurrent aggregation ([#9248](https://github.com/opensearch-project/OpenSearch/pull/9248))
 - Cleanup Unreferenced file on segment merge failure ([#9503](https://github.com/opensearch-project/OpenSearch/pull/9503))
 - [Remote Store] Add support for Remote Translog Store upload stats in `_nodes/stats/` API ([#8908](https://github.com/opensearch-project/OpenSearch/pull/8908))
+- PrimaryShardAllocator refactor to abstract out shard state and method calls ([#9760](https://github.com/opensearch-project/OpenSearch/pull/9760)))
```

Comment:
I don't think this will need a change log. Let's wait for a maintainer to confirm or add skip-changelog tag.

Reply:
Removed the changelog entry in latest commit. Need help from maintainers to add the skip-changelog tag.